src/dal/submissions.ts

```diff
@@ -1,9 +1,42 @@
 import {pool} from './connection';
+import {logging} from '../common/logging';
+import {Submission, verifySubmission} from '../models/submission';
 
 async function getCount(): Promise<number> {
   const query = 'SELECT count(*) as count FROM submissions';
   const [rows] = await pool.query(query);
   return rows[0].count;
 }
 
-export {getCount};
+async function create(submission: Submission): Promise<Submission> {
+  verifySubmission(submission);
+
+  const {
+    latitude, longitude, accuracy, actual_down_speed,
+    actual_upload_speed, testing_for, address, zip_code,
+    provider, connected_with, monthly_price,
+    provider_down_speed, rating, ping, hostname,
+  } = submission;
+
+  const query: string = 'INSERT INTO submissions ' +
+    '(latitude, longitude, accuracy, actual_down_speed, ' +
+    'actual_upload_speed, testing_for, address, zip_code, ' +
+    'provider, connected_with, monthly_price, ' +
+    'provider_down_speed, rating, ping, hostname) ' +
+    'VALUES (? ,? ,? ,? ,? ,? ,? ,? ,? ,? ,? ,? ,? ,? ,?) ';
+
+  const result = await pool.query(
+      query,
+      [latitude, longitude, accuracy, actual_down_speed,
+        actual_upload_speed, testing_for, address, zip_code,
+        provider, connected_with, monthly_price,
+        provider_down_speed, rating, ping, hostname],
+      (err) => {
```

Comment:
I don't think function does anything, remove it? It's a callback pattern and we are using promises with the await.